web/jobs/jobs.go

```diff
@@ -242,16 +245,64 @@ func getTriggerState(c echo.Context) error {
 	if err != nil {
 		return wrapJobsError(err)
 	}
+
 	if err = middlewares.Allow(c, webpermissions.GET, t); err != nil {
-		return err
+		// For konnector trigger, we have a specific logic to check for a permissions
+		if ok := extractKonnectorPermissions(c, instance, t); !ok {
+			return middlewares.ErrForbidden
+		}
 	}
+
 	state, err := jobs.GetTriggerState(t)
 	if err != nil {
 		return wrapJobsError(err)
 	}
 	return jsonapi.Data(c, http.StatusOK, apiTriggerState{t: t.Infos(), s: state}, nil)
 }
 
+func extractKonnectorPermissions(c echo.Context, i *instance.Instance, t jobs.Trigger) (ok bool) {
+	if t.Infos().WorkerType != "konnector" {
+		return
+	}
+	reqDoc, err := middlewares.GetPermission(c)
+	if err != nil {
+		return
+	}
+	var msg struct {
+		Konnector string `json:"konnector"`
+	}
+	if err = t.Infos().Message.Unmarshal(&msg); err != nil {
+		return
+	}
+	var man *apps.KonnManifest
+	if man, err = apps.GetKonnectorBySlug(i, msg.Konnector); err != nil {
+		return
+	}
+	reqRules := reqDoc.Permissions
+	appRules := man.Permissions()
+	return intersectPermissions(reqRules, appRules)
+}
+
+func intersectPermissions(rules1, rules2 permissions.Set) bool {
+	doctypeBlacklist := []string{consts.Files, consts.Settings}
+	// This rule intersection only cross permissions on whole doctypes (no
+	// values).
+	var wholeDoctypes map[string]struct{}
```

Comment:
The linter says you have to initialize this map: https://travis-ci.org/cozy/cozy-stack/jobs/424207646#L984